src/transformers/models/clip/modeling_clip.py

```diff
@@ -347,10 +421,13 @@ def forward(self, hidden_states: torch.Tensor) -> torch.Tensor:
 
 
 class CLIPEncoderLayer(nn.Module):
-    def __init__(self, config: CLIPConfig):
+    def __init__(self, config: CLIPConfig, is_causal=False):
         super().__init__()
         self.embed_dim = config.hidden_size
-        self.self_attn = CLIPAttention(config)
+        if config._attn_implementation == "sdpa":
+            self.self_attn = CLIP_ATTENTION_CLASSES[config._attn_implementation](config, is_causal=is_causal)
+        else:
+            self.self_attn = CLIP_ATTENTION_CLASSES[config._attn_implementation](config)
```

Comment:
We don't use causal masking in the vision tower of CLIP hence this conditioning.